### PR TITLE
API Implement namespaces for SS4

### PR DESCRIPTION
**Pull request by robbieaverill**
I'll run a PSR-2 linter after this is merged to keep the changes easy to read

### CODE REVIEW

File: src/PageTypes/BaseHomePage.php [hunk 1]
```diff
@@ -50,7 +83,7 @@ public function getCMSFields() {
 				TreeDropdownField::create(
 					'LearnMorePageID',
 					_t('BaseHomePage.LearnMoreLink','Page to link the "Learn More" button to:'),
```
Comment (NightJar) [inline src/PageTypes/BaseHomePage.php]: Translations still unaltered

File: src/PageTypes/BaseHomePage.php [hunk 2]
```diff
@@ -62,14 +95,14 @@ public function getCMSFields() {
...
 			$gridConfig = $gridField->getConfig();
-			$gridConfig->getComponentByType('GridFieldAddNewButton')->setButtonName(
+			$gridConfig->getComponentByType(GridFieldAddNewButton::class)->setButtonName(
 				_t('BaseHomePage.AddNewButton','Add new')
 			);
-			$gridConfig->removeComponentsByType('GridFieldAddExistingAutocompleter');
-			$gridConfig->removeComponentsByType('GridFieldDeleteAction');
+			$gridConfig->removeComponentsByType(GridFieldAddExistingAutocompleter::class);
+			$gridConfig->removeComponentsByType(GridFieldDeleteAction::class);
 			$gridConfig->addComponent(new GridFieldDeleteAction());
```
Comment (NightJar) [inline src/PageTypes/BaseHomePage.php]: I believe they `use Injectable` right, so should we change this from `new ` to `::create()`?

File: src/PageTypes/DatedUpdateHolder.php [hunk 3]
```diff
@@ -139,7 +190,7 @@ public static function ExtractMonths(DataList $updates, $link = null, $currentYe
 		$dates = $updates->dataQuery()
 			->groupby('YEAR("Date")')
 			->groupby('MONTH("Date")')
-			->sort('Date', 'DESC')
+			->sort(DBDate::class, 'DESC')
```
Comment (NightJar) [inline src/PageTypes/DatedUpdateHolder.php]: Column name, not class name - I'm going to assume the column hasn't been changed to be named `SilverStripe\Model\...` etc.

File: src/PageTypes/DatedUpdateHolder.php [hunk 4]
```diff
@@ -322,13 +373,13 @@ public function parseParams($produceErrorMessages = true) {
 		if (isset($tag)) $tag = (int)$tag;
 		if (isset($from)) {
 			$from = urldecode($from);
-			$parser = new SS_Datetime;
+			$parser = new DBDatetime;
```
Comment (NightJar) [inline src/PageTypes/DatedUpdateHolder.php]: `DBField::create_field()` instead?

File: src/PageTypes/DatedUpdatePage.php [hunk 5]
```diff
@@ -24,13 +42,13 @@ public function populateDefaults() {
...
 		if(!isset($this->Date) || $this->Date === null) {
-			$this->Date = SS_Datetime::now()->Format('Y-m-d 09:00:00');
+			$this->Date = DBDatetime::now()->Format('Y-m-d 09:00:00');
 		}
 	}
 
 	public function fieldLabels($includerelations = true) {
 		$labels = parent::fieldLabels($includerelations);
-		$labels['Date'] = _t('DateUpdatePage.DateLabel', 'Date');
+		$labels[DBDate::class] = _t('DateUpdatePage.DateLabel', DBDate::class);
```
Comment (NightJar) [inline src/PageTypes/DatedUpdatePage.php]: Again I think the auto magicker has got it a little wrong.

File: src/PageTypes/DatedUpdatePage.php [hunk 6]
```diff
@@ -60,5 +78,5 @@ public function getCMSFields() {
 	}
 }
 
-class DatedUpdatePage_Controller extends Page_Controller {
+class DatedUpdatePage_Controller extends PageController {
```
Comment (NightJar) [inline src/PageTypes/DatedUpdatePage.php]: Needs to be a separate file

File: src/PageTypes/BaseHomePage.php (hunk 1 above)
Comment (robbieaverill) [reply src/PageTypes/BaseHomePage.php]: I might do this a bit further down the line if that's OK?

File: src/PageTypes/BaseHomePage.php (hunk 2 above)
Comment (robbieaverill) [reply src/PageTypes/BaseHomePage.php]: They aren't in SS4. I suppose we could future proof if for SS5 by loading with injector now anyway though

File: src/PageTypes/EventHolder.php [hunk 7]
```diff
@@ -37,7 +48,7 @@ class EventHolder extends DatedUpdateHolder {
 	public static function AllUpdates($className = 'Events', $parentID = null, $tagID = null, $dateFrom = null,
 		$dateTo = null, $year = null, $monthNumber = null) {
 
-		return parent::AllUpdates($className, $parentID, $tagID, $dateFrom, $dateTo, $year, $monthNumber)->Sort('Date', 'ASC');
+		return parent::AllUpdates($className, $parentID, $tagID, $dateFrom, $dateTo, $year, $monthNumber)->Sort(DBDate::class, 'ASC');
```
Comment (NightJar) [inline src/PageTypes/EventHolder.php]: Again with the field name

File: src/PageTypes/EventPage.php [hunk 8]
```diff
@@ -52,11 +77,11 @@ public function populateDefaults() {
 
 	public function getCMSFields() {
 		$this->beforeUpdateCMSFields(function (FieldList $fields) {
-			$fields->removeByName('Date');
+			$fields->removeByName(DBDate::class);
```
Comment (NightJar) [inline src/PageTypes/EventPage.php]: Field name is not a Class Name

File: src/PageTypes/EventPage.php [hunk 9]
```diff
@@ -52,11 +77,11 @@ public function populateDefaults() {
...
 
 			$dateTimeFields = array();
 
-			$dateTimeFields[] = $dateField = DateField::create('Date', 'Date');
+			$dateTimeFields[] = $dateField = DateField::create(DBDate::class, DBDate::class);
```
Comment (NightJar) [inline src/PageTypes/EventPage.php]: Field name.

File: src/PageTypes/EventPage.php [hunk 10]
```diff
@@ -82,6 +107,6 @@ public function NiceLocation() {
 	}
 }
 
-class EventPage_Controller extends Page_Controller {
+class EventPage_Controller extends PageController {
```
Comment (NightJar) [inline src/PageTypes/EventPage.php]: Separate class

File: src/PageTypes/NewsHolder.php [hunk 11]
```diff
@@ -34,10 +45,10 @@ class NewsHolder extends DatedUpdateHolder {
 	 *
 	 * @returns DataList | PaginatedList
 	 */
-	public static function AllUpdates($className = 'NewsPage', $parentID = null, $tagID = null, $dateFrom = null,
+	public static function AllUpdates($className = NewsPage::class, $parentID = null, $tagID = null, $dateFrom = null,
 			$dateTo = null, $year = null, $monthNumber = null) {
 
-		return parent::AllUpdates($className, $parentID, $tagID, $dateFrom, $dateTo, $year, $monthNumber)->Sort('Date', 'DESC');
+		return parent::AllUpdates($className, $parentID, $tagID, $dateFrom, $dateTo, $year, $monthNumber)->Sort(DBDate::class, 'DESC');
```
Comment (NightJar) [inline src/PageTypes/NewsHolder.php]: Date field name (c.f. sort)

File: src/PageTypes/SitemapPage.php [hunk 12]
```diff
@@ -8,7 +22,7 @@ class SitemapPage extends Page {
 	private static $plural_name = 'Sitemap Pages';
 
 }
-class SitemapPage_Controller extends Page_Controller {
+class SitemapPage_Controller extends PageController {
```
Comment (NightJar) [inline src/PageTypes/SitemapPage.php]: Underscore shouldn't be a thing in the class name, not just the extension.

Also it should be in another file.

File: src/Search/CwpSearchPage.php [hunk 13]
```diff
@@ -21,7 +32,7 @@ public function canViewStage($stage = 'Live', $member = null) {
  *
  * @author dmooyman
  */
-class CwpSearchPage_Controller extends Page_Controller {
+class CwpSearchPage_Controller extends PageController {
```
Comment (NightJar) [inline src/Search/CwpSearchPage.php]: remove `_` and move definition to another file/namespace.

File: src/Tasks/CleanupGeneratedPdfDailyTask.php [hunk 14]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace CWP\CWP\Tasks;
+
+use DailyTask;
+
+
+class CleanupGeneratedPdfDailyTask extends DailyTask {
```
Comment (NightJar) [inline src/Tasks/CleanupGeneratedPdfDailyTask.php]: DailyTask is deprecated, as per #9

**Comment by robbieaverill** [comment]
Ok, updated CwpSearchPage

**Pull request merged**